### PR TITLE
Added Configuration service

### PR DESCRIPTION
**Pull request by spoenemann**
This is a first step towards #22.

I also renamed the JWT_PRIVATE_KEY option to OCT_JWT_PRIVATE_KEY for consistency.

### CODE REVIEW

File: packages/open-collaboration-server/src/auth-endpoints/oauth-endpoint.ts [hunk 1]
```diff
@@ -65,11 +68,12 @@ export abstract class OAuthEndpoint implements AuthEndpoint {
                     res.send('Internal server error occured during Login. Please try again');
                     return;
                 }
-                if(process.env.OCT_LOGIN_SUCCESS_URL) {
-                    res.redirect(process.env.OCT_LOGIN_SUCCESS_URL);
+                const loginSuccessURL = this.configuration.getValue('oct-login-success-url');
```
Comment (kaisalmen) [inline packages/open-collaboration-server/src/auth-endpoints/oauth-endpoint.ts:71]: I would put the known keys in some map or enum and reference that. The Kebap transformation can then become optional sometimes.

**Review by kaisalmen — COMMENTED** [review]
Good idea. I have some suggestions, nothing blocking, though.

**Comment by kaisalmen** [comment]
@spoenemann and reminder to myself, we must change the private key identifier in the public deployment.

File: packages/open-collaboration-server/src/utils/configuration.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+export const Configuration = Symbol('Configuration');
+
+@injectable()
+export class DefaultConfiguration implements Configuration {
+
+    getValue(key: string, type?: 'string'): string | undefined;
+    getValue(key: string, type: 'number'): number | undefined;
+    getValue(key: string, type: 'boolean'): boolean | undefined;
+    getValue(key: string, type?: 'string' | 'number' | 'boolean'): string | number | boolean | undefined {
```
[the remark this answers is not in the document]
Comment (spoenemann) [reply packages/open-collaboration-server/src/utils/configuration.ts:39]: What do you mean exactly? Would you rather have separate methods `getNumberValue` etc.?

[the remark this answers is not in the document]
Comment (kaisalmen) [reply packages/open-collaboration-server/src/utils/configuration.ts:39]: Three methods like the interface definitions without any branching needed inside

[the remark this answers is not in the document]
Comment (spoenemann) [reply packages/open-collaboration-server/src/utils/configuration.ts:39]: It's not possible to have separate methods unless we give them separate names. There's no method overloading in JS.

**Review by msujew — APPROVED** [review]
Looks good to me 👍

File: packages/open-collaboration-server/src/utils/configuration.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (kaisalmen) [reply packages/open-collaboration-server/src/utils/configuration.ts:39]: Sorry, I tend to forget that. Should the interface be reduced to one method then as well?

**Review by kaisalmen — APPROVED** [review]
Let's go.

File: packages/open-collaboration-server/src/utils/configuration.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (msujew) [reply packages/open-collaboration-server/src/utils/configuration.ts:39]: No, TypeScript is still aware of overloads, and given that these signatures return different types based on the input, reducing all of them to a single signature would make it difficult to deal with the return value. It's only JavaScript that is unaware of overloads.

[the remark this answers is not in the document]
Comment (kaisalmen) [reply packages/open-collaboration-server/src/utils/configuration.ts:39]: Thanks, I just checked out the branch and this becomes clear when you see the compiler errors. I guess there is no way to automatically handle this "overload"-reduction by `tsc`.

**Pull request merged**